pkg/otelcollector/infra-meter/infra-meter.go

```diff
@@ -231,3 +257,68 @@ func updateK8sAttributesProcessor(processor interface{}, selectorsList []interfa
 		}
 	}
 }
+
+// processMapForSecrets processes the given map to search for the secret reference and replace it with actual value.
+func processMapForSecrets(config map[string]interface{}, secretManagerClient *secretmanager.SecretManagerClient) error {
+	for key, value := range config {
+		switch v := value.(type) {
+		case map[string]interface{}:
+			if err := processMapForSecrets(v, secretManagerClient); err != nil {
+				return fmt.Errorf("error processing map for key %s: %w", key, err)
+			}
+		case string:
+			if strings.HasPrefix(v, "${") && strings.HasSuffix(v, "}") {
+				secretValue, err := getSecretValue(v, secretManagerClient)
+				if err != nil {
+					return fmt.Errorf("error getting secret value for key %s: %w", key, err)
+				}
+				config[key] = secretValue
+			}
+		case []interface{}:
+			for i, val := range v {
+				switch valStr := val.(type) {
+				case map[string]interface{}:
+					if err := processMapForSecrets(valStr, secretManagerClient); err != nil {
+						return fmt.Errorf("error processing map in slice for key %s: %w", key, err)
+					}
+				case string:
+					if strings.HasPrefix(valStr, "${") && strings.HasSuffix(valStr, "}") {
+						secretValue, err := getSecretValue(valStr, secretManagerClient)
+						if err != nil {
+							return fmt.Errorf("error getting secret value in slice for key %s: %w", key, err)
+						}
+						v[i] = secretValue
+					}
+				}
+			}
+			config[key] = v
+		}
+	}
+
+	return nil
+}
+
+// getSecretValue fetches secret value from the GCP Secret Manager.
+func getSecretValue(valueString string, secretManagerClient *secretmanager.SecretManagerClient) (string, error) {
+	secretName := strings.TrimPrefix(valueString, "${")
+	secretName = strings.TrimSuffix(secretName, "}")
+
+	secretName = fmt.Sprintf("projects/%s/secrets/%s-%s/versions/latest", secretManagerClient.GCPProjectID, secretManagerClient.ProjectID, secretName)
+
+	accessRequest := &secretmanagerpb.AccessSecretVersionRequest{
+		Name: secretName,
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
```

Comment:
Do we need this defer here? By the time we're out of the function the request will be finished (failed or success)

Reply:
I think we need this here as a Golang practise for context.WithTimeout when the request is completed or timed out as this is a separate context for making calls to GCP